### PR TITLE
Use `timedelta` for test timeouts

### PR DESCRIPTION
**Pull request by zpcore**
# Description

Split the `update data type into timestamp instead of primitives` out from https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/263. 

We follow with our internal style guide recommends to use datetime et al instead of primitives: https://engdoc.corp.google.com/eng/doc/devguide/py/totw/047.md?cl=head

# Tests

[mxla_maxtext_nightly_gke
](https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/mxla_maxtext_nightly_gke/grid)

[jax_integration](https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/jax_integration/grid)

[flax_latest_supported](https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/flax_latest_supported/grid)

[pytorchxla-nightly](https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/pytorchxla-nightly/grid)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: xlml/apis/test_config.py [hunk 1]
```diff
@@ -400,8 +402,7 @@ def _from_json_helper(
         setup=setup,
         exports=exports,
         test_command=test_command,
-        # `timeout` is in seconds
-        time_out_in_min=test['timeout'] // 60,
+        time_out=datetime.timedelta(minutes=test['timeout']),
```
Comment (RissyRan) [inline xlml/apis/test_config.py:405]: this is seconds? From JSonnet. Please check change below as well.

Comment (zpcore) [reply xlml/apis/test_config.py:405]: You are right, thanks for catching that! Updated.

**Review by RissyRan — APPROVED** [review]
LGTM! Please merge on Monday.

File: xlml/apis/task.py [hunk 2]
```diff
@@ -397,7 +395,7 @@ def run_model(
         )
       launch_workload = self.launch_workload(workload_id, gcs_path)
       wait_for_workload_completion = xpk.wait_for_workload_completion.override(
-          timeout=self.task_test_config.time_out_in_min * 60,
+          timeout=int(self.task_test_config.time_out.total_seconds()),
```
Comment (will-cromar) [inline xlml/apis/task.py:398]: I have the tiniest nit: Airflow uses `timeout` instead of `time_out`, so please do the same with out field.

**Review by will-cromar — COMMENTED** [review]
Otherwise LGTM. Thanks!

**Review by will-cromar — APPROVED** [review]

File: xlml/apis/task.py (hunk 2 above)
Comment (zpcore) [reply xlml/apis/task.py:398]: Do you mean use `timeout` instead of `time_out`?

Comment (will-cromar) [reply xlml/apis/task.py:398]: Yeah, just to keep our naming consistent

File: xlml/utils/metric_test.py [hunk 3]
```diff
@@ -346,7 +347,7 @@ def test_add_test_config_metadata(self):
         test_name="test_name",
         set_up_cmds="set_up_cmds",
         run_model_cmds="run_model_cmds",
-        time_out_in_min=60,
+        timeout=datetime.timedelta(minutes=60),
```
Comment (vanbasten23) [inline xlml/utils/metric_test.py:350]: Could you leave a comment in the pr description of what the intention for using dattime.timedelta?

**Review by vanbasten23 — APPROVED** [review]

File: xlml/utils/metric_test.py (hunk 3 above)
Comment (zpcore) [reply xlml/utils/metric_test.py:350]: Added in the description.

**Comment by will-cromar** [comment]
Bypassing code owners since this is a no-op.

**Pull request merged**